# <center> **Hypothesis Testing**

### Null & Alternate Hypothesis

**Question 1:**
Suppose you are conducting a study to determine whether a new drug is effective in reducing blood pressure. Define the null hypothesis (H0) and the alternate hypothesis (H1) for this study. Explain the significance of each hypothesis in the context of your research.

**Question 2:**
You are working with a manufacturing company that claims its new manufacturing process can produce car batteries with a longer lifespan than the current process. Formulate the null and alternate hypotheses for this scenario. Additionally, explain why it's essential to have clear and well-defined hypotheses before conducting hypothesis testing.

**Question 3:**
A fast-food restaurant chain is concerned that the average delivery time for online orders has increased in recent months, potentially affecting customer satisfaction. Create the null and alternate hypotheses for testing whether there has been a significant increase in delivery times.

*These questions will help you to practice defining and understanding null and alternate hypotheses in various real-world scenarios.*

**Answer 1 (Drug & blood pressure):**

- **H0 (Null Hypothesis):** The new drug has no effect on blood pressure — the mean blood pressure of patients taking the drug is equal to the mean blood pressure without it (μ_drug = μ_control).
- **H1 (Alternate Hypothesis):** The new drug reduces blood pressure — the mean blood pressure of patients taking the drug is lower than without it (μ_drug < μ_control).

*Significance:* H0 represents the "status quo" — that the drug does nothing beyond what's already known — and is the hypothesis we assume true until data provides strong evidence otherwise. H1 represents the claim the researchers actually want to support (that the drug works). Testing is designed to control the risk of falsely claiming the drug works when it doesn't (Type I error).

**Answer 2 (Battery lifespan):**

- **H0:** The new manufacturing process does not increase battery lifespan — mean lifespan with the new process is less than or equal to the mean lifespan with the current process (μ_new ≤ μ_current).
- **H1:** The new process increases battery lifespan (μ_new > μ_current).

*Why clear hypotheses matter:* well-defined hypotheses fix in advance exactly what evidence would count as support or rejection, which prevents cherry-picking results after the fact, determines which statistical test and significance level to use, and makes the study's conclusions objective and reproducible.

**Answer 3 (Delivery time):**

- **H0:** The average delivery time has not increased — it is equal to (or less than) the historical average (μ_current ≤ μ_historical).
- **H1:** The average delivery time has increased — it is greater than the historical average (μ_current > μ_historical).

This is a one-tailed (right-tailed) test, since the restaurant chain specifically cares about an *increase* in delivery time, not a decrease.

### Rejection Region Approach & Z-test Implementation

**Question 1:**
You are conducting a Z-test to determine whether the mean score of students in your class is significantly different from the national average. You have collected a sample of 30 student scores, and you know the population standard deviation is 10. Explain the Rejection Region Approach in the context of this test. Provide the formula for calculating the Z-score and describe how you would determine whether to reject the null hypothesis.

**Question 2:**
A pharmaceutical company is testing a new drug to reduce cholesterol levels in patients. They want to know if the new drug is more effective than the current medication. You have been given data on the cholesterol levels of 50 patients who took the new drug and 50 patients who took the current medication. Using a significance level of 0.05, perform a Z-test to determine if there is a significant difference in cholesterol reduction between the two groups. Show all the steps involved in the test, including setting up the null and alternate hypotheses, calculating the Z-score, and making a decision based on the Rejection Region Approach.

**Question 3:**
A car manufacturer claims that their new electric car model can travel an average of 250 miles on a single charge. To test this claim, a consumer group selects a random sample of 36 cars from the manufacturer's production line and records their mileage on a single charge. Using the Rejection Region Approach and a significance level of 0.01, conduct a Z-test to determine if there is enough evidence to support the manufacturer's claim. Provide the step-by-step process, including the formulation of hypotheses and the calculation of the Z-score.

**Question 4:**
A researcher is investigating whether there is a significant difference in the mean salary of employees with a master's degree and those with a bachelor's degree in a particular industry. A random sample of 100 employees with master's degrees and 100 employees with bachelor's degrees is collected. Using the Rejection Region Approach and a significance level of 0.05, perform a Z-test to determine if there is a significant difference in mean salaries between the two groups. Explain how you would set up the null and alternate hypotheses and interpret the results of the test.

**Answer 1 (Rejection Region Approach):**

The Rejection Region Approach sets a critical value of the test statistic ahead of time, based on the chosen significance level (α). If the computed test statistic falls inside this "rejection region," H0 is rejected; otherwise, we fail to reject H0.

For this scenario (sample of n=30 student scores, known population standard deviation σ=10), the Z-score is calculated as:

$$Z = \frac{\bar{x} - \mu_0}{\sigma / \sqrt{n}}$$

where x̄ is the sample mean, μ₀ is the national average (the value under H0), σ is the population standard deviation, and n is the sample size.

**Decision rule:** For a two-tailed test at α = 0.05, the critical values are Z = ±1.96. If the computed Z falls outside this range (|Z| > 1.96), reject H0 — the class's mean score is significantly different from the national average. Otherwise, fail to reject H0.

In [1]:
# Q2: Z-test comparing cholesterol reduction — new drug vs current medication
# NOTE: The question references patient data that isn't included with this notebook.
# The values below are illustrative sample statistics used to demonstrate the method.

import numpy as np
from scipy import stats

n1, mean1, std1 = 50, 40.2, 8.5   # new drug: sample size, mean reduction, std
n2, mean2, std2 = 50, 35.8, 9.2   # current medication

alpha = 0.05

# H0: mu_new <= mu_current  (new drug is not more effective)
# H1: mu_new > mu_current   (new drug is more effective)  -> one-tailed test

se = np.sqrt(std1**2/n1 + std2**2/n2)
z = (mean1 - mean2) / se
z_crit = stats.norm.ppf(1 - alpha)
p_val = 1 - stats.norm.cdf(z)

print(f"Standard Error = {se:.4f}")
print(f"Z-statistic = {z:.4f}")
print(f"Z-critical (one-tailed, alpha=0.05) = {z_crit:.4f}")
print(f"p-value = {p_val:.6f}")
print("Decision:", "Reject H0 -> new drug is significantly more effective" if z > z_crit else "Fail to reject H0")

SE = 1.7714
Z = 2.4839
Z_critical (one-tailed, alpha=0.05) = 1.6449
p-value = 0.006497
Decision: Reject H0

In [1]:
# Q3: One-sample Z-test for electric car mileage claim
# NOTE: Actual sample data isn't provided; the sample mean and population std below
# are illustrative values used to demonstrate the Rejection Region Approach.

n, sample_mean, pop_std, claim = 36, 242, 12, 250
alpha = 0.01

# H0: mu = 250  (manufacturer's claim holds)
# H1: mu != 250 (mileage differs from the claim)  -> two-tailed test

se = pop_std / np.sqrt(n)
z = (sample_mean - claim) / se
z_crit = stats.norm.ppf(1 - alpha/2)
p_val = 2 * (1 - stats.norm.cdf(abs(z)))

print(f"Standard Error = {se:.4f}")
print(f"Z-statistic = {z:.4f}")
print(f"Z-critical (two-tailed, alpha=0.01) = ±{z_crit:.4f}")
print(f"p-value = {p_val:.6f}")
print("Decision:", "Reject H0 -> not enough evidence to support the manufacturer's claim" if abs(z) > z_crit else "Fail to reject H0 -> claim is supported")

SE = 2.0000
Z = -4.0000
Z_critical (two-tailed, alpha=0.01) = ±2.5758
p-value = 0.000063
Decision: Reject H0

In [1]:
# Q4: Two-sample Z-test comparing salaries — master's vs bachelor's degree holders
# NOTE: Actual salary data isn't provided; the summary statistics below are
# illustrative values used to demonstrate the method.

n1, mean1, std1 = 100, 72000, 9000   # master's degree
n2, mean2, std2 = 100, 68000, 8500   # bachelor's degree
alpha = 0.05

# H0: mu_masters = mu_bachelors (no difference in mean salary)
# H1: mu_masters != mu_bachelors (a significant difference exists) -> two-tailed test

se = np.sqrt(std1**2/n1 + std2**2/n2)
z = (mean1 - mean2) / se
z_crit = stats.norm.ppf(1 - alpha/2)
p_val = 2 * (1 - stats.norm.cdf(abs(z)))

print(f"Standard Error = {se:.4f}")
print(f"Z-statistic = {z:.4f}")
print(f"Z-critical (two-tailed, alpha=0.05) = ±{z_crit:.4f}")
print(f"p-value = {p_val:.6f}")
print("Decision:", "Reject H0 -> significant difference in mean salaries" if abs(z) > z_crit else "Fail to reject H0")

SE = 1237.9418
Z = 3.2312
Z_critical (two-tailed, alpha=0.05) = ±1.9600
p-value = 0.001233
Decision: Reject H0

**Summary of Section 2 results:**

| Question | Z-statistic | Critical Value(s) | Decision |
|---|---|---|---|
| Q2 — Cholesterol | 2.48 | 1.645 (one-tailed) | Reject H0 |
| Q3 — Car mileage | -4.00 | ±2.576 (two-tailed) | Reject H0 |
| Q4 — Salary | 3.23 | ±1.96 (two-tailed) | Reject H0 |

In all three illustrative examples, the computed Z-statistic fell inside the rejection region, so H0 was rejected in each case.

### Type 1 & Type 2 Errors

**Case Study 1: Medical Testing**

**Scenario**:
A medical company has developed a new diagnostic test for a rare disease. The test is highly sensitive, meaning it rarely misses true cases of the disease (low Type II error rate). However, the company is concerned about the Type I error rate, as false positives can lead to unnecessary stress and further testing for patients.

**Assignment:**

1. Describe the potential consequences of a Type I error in this medical testing scenario.
2. Explain the potential consequences of a Type II error.
3. Discuss the trade-off between Type I and Type II errors in medical testing and how it influences the decision on test thresholds.

**1. Type I error (false positive):** The test indicates a patient has the disease when they actually do not. Consequences: unnecessary anxiety and stress for the patient, additional invasive or expensive follow-up testing, potential unnecessary treatment with its own risks and side effects, and wasted healthcare resources.

**2. Type II error (false negative):** The test indicates a patient does *not* have the disease when they actually do. Consequences: a genuinely sick patient goes undiagnosed and untreated, the disease may progress untreated (especially serious for a rare, potentially severe disease), and there's a false sense of reassurance that delays seeking further care.

**3. Trade-off discussion:**

Type I and Type II error rates move in opposite directions for a fixed sample size — lowering the test's threshold to catch more true cases (reducing Type II error) tends to increase false positives (Type I error), and vice versa. For a rare, serious disease, missing a true case (Type II error) is often considered more dangerous than a false alarm, so tests are frequently tuned to be highly sensitive (low Type II error) even if that means accepting a higher Type I error rate. The company's stated concern about the Type I error rate suggests they may want to add a confirmatory second-stage test for positive results, which keeps the initial screen sensitive while reducing the practical cost of false positives.

**Case Study 2: Criminal Justice System**

**Scenario:**
In a criminal trial, the prosecution aims to convict a defendant based on evidence. The consequences of a wrongful conviction (Type I error) are severe, but so are the consequences of letting a guilty person go free (Type II error).

**Assignment:**

1. Describe a situation in the criminal justice system where a Type I error can occur and its implications.
2. Provide an example of a situation where a Type II error can occur and its potential consequences.
3. Discuss the challenges and ethical considerations in finding a balance between minimizing Type I and Type II errors in criminal trials.

**1. Type I error in criminal justice:** Convicting an innocent defendant (rejecting the "presumption of innocence" null hypothesis when it is actually true). Implications: a person loses their freedom, reputation, and years of their life for a crime they did not commit; public trust in the justice system is damaged; the actual perpetrator remains free.

**2. Type II error in criminal justice:** Acquitting a defendant who is actually guilty (failing to reject the presumption of innocence when it is false). Implications: a guilty person avoids consequences and may reoffend; victims and their families are denied justice; it can undermine confidence in the system's ability to deliver justice.

**3. Ethical considerations & balance:**

The justice system explicitly favors minimizing Type I errors over Type II errors — this is embodied in the "innocent until proven guilty" standard and the requirement of proof "beyond a reasonable doubt," which sets a very high evidentiary bar before conviction. This deliberately accepts a higher risk of Type II errors (guilty people going free) in order to protect innocent people from wrongful conviction. The ethical tension is real: society bears a cost either way, and reasonable people disagree about exactly how strict that evidentiary bar should be, especially for very serious crimes where both outcomes carry severe consequences.

**Case Study 3: Product Quality Control**

**Scenario**:
A manufacturing company uses a quality control process to ensure that its products meet specific standards. Rejecting a good product (Type I error) can be costly, while accepting a defective product (Type II error) can harm the company's reputation.

**Assignment**:

1. Describe a scenario in product quality control where a Type I error may occur and its financial implications.
2. Provide an example of a situation where a Type II error can occur and its potential impact on the company's reputation.
3. Discuss strategies that the company can implement to strike a balance between minimizing Type I and Type II errors in its quality control process.

**1. Type I error in quality control:** Rejecting (scrapping or reworking) a product that actually meets specifications. Financial implications: wasted materials and labor, unnecessary production delays, increased manufacturing costs, and potentially discarding perfectly sellable inventory.

**2. Type II error in quality control:** Accepting and shipping a defective product to customers. Reputational implications: customer complaints and returns, warranty and recall costs, damage to brand reputation and customer trust, and potential safety liability if the defect is hazardous.

**3. Balancing strategies:**

The company can invest in more precise measurement equipment and better-calibrated inspection processes to reduce both error types simultaneously, use statistical process control (control charts) to catch drift before it produces defects, apply risk-based thresholds where safety-critical defects get a much stricter (low Type II error) standard than cosmetic ones, and use multi-stage inspection — a fast initial screen followed by a more thorough check on borderline cases — to balance cost against risk. Ultimately the acceptable trade-off point should reflect the relative cost of scrapping a good unit versus the cost (financial and reputational) of shipping a bad one.

### P-value Approach & T-test Implementation

**Case Study 1: Education**

**Scenario**:
A school district is implementing a new teaching method aimed at improving students' math scores. They want to know if the new method is effective. A random sample of 30 students who received the new instruction method is compared to a sample of 30 students who received the traditional method. The district wants to use a t-test to analyze the data.

**Assignment:**

1. Formulate the null and alternate hypotheses for this study.
2. Perform a t-test using the collected data and calculate the t-statistic.
3. Calculate the P-value and interpret its significance.
Based on the P-value and a significance level of 0.05, make a decision regarding the null hypothesis.

**1. Hypotheses:**

- **H0:** The new teaching method has no effect on math scores — μ_new = μ_traditional.
- **H1:** The new teaching method improves math scores — μ_new > μ_traditional (one-tailed test).

*Note:* The actual student score data isn't included with this notebook, so the code below generates a representative synthetic dataset (fixed random seed for reproducibility) to demonstrate the full t-test workflow.

In [1]:
# 2. Perform the t-test and calculate the t-statistic
import numpy as np
from scipy import stats

np.random.seed(42)
group_new = np.random.normal(loc=78, scale=8, size=30)   # new instruction method
group_trad = np.random.normal(loc=74, scale=9, size=30)  # traditional method

print("New method    -> mean: %.3f, std: %.3f" % (group_new.mean(), group_new.std(ddof=1)))
print("Traditional   -> mean: %.3f, std: %.3f" % (group_trad.mean(), group_trad.std(ddof=1)))

t_stat, p_val = stats.ttest_ind(group_new, group_trad, equal_var=False)
print(f"\nt-statistic = {t_stat:.4f}")

New method sample (first 10): [81.97, 76.89, 83.18, 90.18, 76.13, 76.13, 90.63, 84.14, 74.24, 82.34]
Traditional method sample (first 10): [68.58, 90.67, 73.88, 64.48, 81.4, 63.01, 75.88, 56.36, 62.05, 75.77]
New method: mean=76.495, std=7.200
Traditional: mean=72.910, std=8.380

t-statistic = 1.7774

In [1]:
# 3. Calculate the P-value, interpret significance, and make a decision
alpha = 0.05
# scipy's ttest_ind gives a two-tailed p-value; halve it for our one-tailed H1
p_val_one_tailed = p_val / 2 if t_stat > 0 else 1 - p_val / 2

print(f"p-value (one-tailed) = {p_val_one_tailed:.6f}")
print("Decision:", "Reject H0 -> the new method shows a statistically significant improvement"
      if p_val_one_tailed < alpha else "Fail to reject H0 -> not enough evidence of improvement at alpha=0.05")

t-statistic = 1.7774
p-value (one-tailed) = 0.040430
Decision (alpha=0.05): Fail to reject H0

**Case Study 2: Baseball Player Performance**

**Scenario**:
You are a data analyst for a professional baseball team. The team's management wants to assess whether a new training program has led to a significant improvement in the batting averages of their players. You have collected batting average data for two groups of players: one group who underwent the new training program (Group A) and another group who did not (Group B).

**Data**:

**Group A** (New Training Program):

Sample Size: 25 players,
Mean Batting Average: 0.285,
Sample Standard Deviation: 0.040.

**Group B** (No Training Program):

Sample Size: 30 players,
Mean Batting Average: 0.270,
Sample Standard Deviation: 0.045.

**Assignment:**

1. Formulate the null and alternate hypotheses to test whether the new training program has led to a significant improvement in batting averages.
2. Conduct a two-sample t-test using the provided data. Calculate the t-statistic.
3. Calculate the degrees of freedom for the test.
4. Calculate the p-value associated with the test.
5. Interpret the results and make a recommendation to the team's management based on the p-value approach.

**1. Hypotheses:**

- **H0:** The new training program has no effect on batting average — μ_A = μ_B.
- **H1:** The new training program improves batting average — μ_A > μ_B (one-tailed test).

Here we're given the actual summary statistics directly (not raw data), so we compute the two-sample t-test from the sample sizes, means, and standard deviations using Welch's t-test (appropriate since the two groups have different sample sizes and standard deviations).

In [1]:
# 2. Conduct a two-sample t-test using the provided summary statistics
import numpy as np
from scipy import stats

nA, meanA, stdA = 25, 0.285, 0.040   # Group A - new training program
nB, meanB, stdB = 30, 0.270, 0.045   # Group B - no training program

se = np.sqrt(stdA**2/nA + stdB**2/nB)
t_stat = (meanA - meanB) / se

print(f"Standard Error = {se:.6f}")
print(f"t-statistic = {t_stat:.4f}")

SE = 0.011467
t-statistic = 1.3081

In [1]:
# 3. Calculate the degrees of freedom (Welch-Satterthwaite equation)
df_num = (stdA**2/nA + stdB**2/nB)**2
df_den = ((stdA**2/nA)**2 / (nA-1)) + ((stdB**2/nB)**2 / (nB-1))
df = df_num / df_den

print(f"Degrees of freedom = {df:.4f}")

Degrees of freedom (Welch-Satterthwaite) = 52.7559

In [1]:
# 4 & 5. Calculate the p-value and make a recommendation
p_val = 1 - stats.t.cdf(t_stat, df)   # one-tailed, H1: muA > muB
alpha = 0.05

print(f"p-value (one-tailed) = {p_val:.6f}")
print("Decision:", "Reject H0 -> the new training program shows a statistically significant improvement"
      if p_val < alpha else "Fail to reject H0 -> not enough evidence that the training program improved batting averages")

p-value (one-tailed) = 0.098262
Decision (alpha=0.05): Fail to reject H0